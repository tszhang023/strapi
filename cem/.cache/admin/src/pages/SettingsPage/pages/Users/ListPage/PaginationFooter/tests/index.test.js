import React from 'react';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import { createMemoryHistory } from 'history';
import { Router, Route } from 'react-router-dom';
import Theme from '../../../../../../../components/Theme';
import PaginationFooter from '../index';

const makeApp = (history, pagination) => {
  return (
    <IntlProvider messages={{}} textComponent="span" locale="en" defaultLocale="en">
      <Theme>
        <Router history={history}>
          <Route path="/settings/user">
            <PaginationFooter pagination={pagination} />
          </Route>
        </Router>
      </Theme>
    </IntlProvider>
  );
};

describe('DynamicTable', () => {
  it('renders and matches the snapshot', () => {
    const history = createMemoryHistory();
    const pagination = { pageCount: 2 };
    history.push('/settings/user?pageSize=10&page=1&sort=firstname');
    const app = makeApp(history, pagination);

    const { container } = render(app);

    expect(container).toMatchSnapshot();
  });
});
