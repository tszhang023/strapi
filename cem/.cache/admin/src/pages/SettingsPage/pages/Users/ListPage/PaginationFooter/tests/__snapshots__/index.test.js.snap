// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DynamicTable renders and matches the snapshot 1`] = `
.c28 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c0 {
  padding-top: 16px;
}

.c14 {
  padding-left: 8px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c17 > * + * {
  margin-left: 4px;
}

.c20 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c23 {
  font-weight: 600;
  color: #32324d;
  font-size: 0.75rem;
  line-height: 1.33;
}

.c26 {
  color: #32324d;
  font-size: 0.75rem;
  line-height: 1.33;
}

.c24 {
  line-height: revert;
}

.c18 {
  padding: 12px;
  border-radius: 4px;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  outline: none;
}

.c18:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c18:focus-visible {
  outline: none;
}

.c18:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c21 {
  padding: 12px;
  border-radius: 4px;
  box-shadow: 0px 1px 4px rgba(33,33,52,0.1);
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  outline: none;
}

.c21:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c21:focus-visible {
  outline: none;
}

.c21:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c22 {
  color: #271fe0;
  background: #ffffff;
}

.c22:hover {
  box-shadow: 0px 1px 4px rgba(33,33,52,0.1);
}

.c25 {
  color: #32324d;
}

.c25:hover {
  box-shadow: 0px 1px 4px rgba(33,33,52,0.1);
}

.c19 {
  font-size: 0.7rem;
  pointer-events: none;
}

.c19 svg path {
  fill: #c0c0cf;
}

.c19:focus svg path,
.c19:hover svg path {
  fill: #c0c0cf;
}

.c27 {
  font-size: 0.7rem;
}

.c27 svg path {
  fill: #666687;
}

.c27:focus svg path,
.c27:hover svg path {
  fill: #4a4a6a;
}

.c6 {
  position: absolute;
  left: 0;
  right: 0;
  bottom: 0;
  top: 0;
  width: 100%;
  background: transparent;
  border: none;
}

.c6:focus {
  outline: none;
}

.c6[aria-disabled='true'] {
  cursor: not-allowed;
}

.c10 {
  color: #32324d;
  display: block;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  font-size: 0.875rem;
  line-height: 1.43;
}

.c9 {
  padding-right: 16px;
  padding-left: 16px;
}

.c11 {
  padding-left: 12px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 > * {
  margin-top: 0;
  margin-bottom: 0;
}

.c3 > * + * {
  margin-top: 0px;
}

.c5 {
  position: relative;
  border: 1px solid #dcdce4;
  padding-right: 12px;
  border-radius: 4px;
  background: #ffffff;
  overflow: hidden;
  min-height: 2rem;
  outline: none;
  box-shadow: 0;
  -webkit-transition-property: border-color,box-shadow,fill;
  transition-property: border-color,box-shadow,fill;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
}

.c5:focus-within {
  border: 1px solid #4945ff;
  box-shadow: #4945ff 0px 0px 0px 2px;
}

.c12 {
  background: transparent;
  border: none;
  position: relative;
  z-index: 1;
}

.c12 svg {
  height: 0.6875rem;
  width: 0.6875rem;
}

.c12 svg path {
  fill: #666687;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: none;
  border: none;
}

.c13 svg {
  width: 0.375rem;
}

.c8 {
  width: 100%;
}

.c15 {
  color: #666687;
  font-size: 0.875rem;
  line-height: 1.43;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div>
          <div
            class="c3"
          >
            <div
              class="c4 c5"
            >
              <button
                aria-disabled="false"
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-label="Entries per page"
                aria-labelledby="select-1-label select-1-content"
                class="c6"
                id="select-1"
                type="button"
              />
              <div
                class="c7 c8"
              >
                <div
                  class="c4"
                >
                  <div
                    class="c9"
                  >
                    <span
                      class="c10"
                      id="select-1-content"
                    >
                      10
                    </span>
                  </div>
                </div>
                <div
                  class="c4"
                >
                  <button
                    aria-hidden="true"
                    class="c11 c12 c13"
                    tabindex="-1"
                    type="button"
                  >
                    <svg
                      fill="none"
                      height="1em"
                      viewBox="0 0 14 8"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M14 .889a.86.86 0 01-.26.625L7.615 7.736A.834.834 0 017 8a.834.834 0 01-.615-.264L.26 1.514A.861.861 0 010 .889c0-.24.087-.45.26-.625A.834.834 0 01.875 0h12.25c.237 0 .442.088.615.264a.86.86 0 01.26.625z"
                        fill="#32324D"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="c14"
        >
          <label
            class="c15"
            for="page-size"
          >
            Entries per page
          </label>
        </div>
      </div>
      <nav
        aria-label="pagination"
        class="sc-fHeRUh"
      >
        <ul
          class="c16 c17"
        >
          <li>
            <a
              aria-current="page"
              aria-disabled="true"
              class="c18 c19 active"
              href="/settings/user"
              tabindex="-1"
            >
              <div
                class="c20"
              >
                Go to previous page
              </div>
              <svg
                aria-hidden="true"
                fill="none"
                height="1em"
                viewBox="0 0 10 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.88 14.12L3.773 8 9.88 1.88 8 0 0 8l8 8 1.88-1.88z"
                  fill="#32324D"
                />
              </svg>
            </a>
          </li>
          <li>
            <a
              aria-current="page"
              class="c21 c22 active"
              href="/settings/user?pageSize=10&page=1&sort=firstname"
            >
              <div
                class="c20"
              >
                Go to page 1
              </div>
              <span
                aria-hidden="true"
                class="c23 c24"
              >
                1
              </span>
            </a>
          </li>
          <li>
            <a
              aria-current="page"
              class="c18 c25 active"
              href="/settings/user?pageSize=10&page=2&sort=firstname"
            >
              <div
                class="c20"
              >
                Go to page 2
              </div>
              <span
                aria-hidden="true"
                class="c26 c24"
              >
                2
              </span>
            </a>
          </li>
          <li>
            <a
              aria-current="page"
              aria-disabled="false"
              class="c18 c27 active"
              href="/settings/user?pageSize=10&page=2&sort=firstname"
            >
              <div
                class="c20"
              >
                Go to next page
              </div>
              <svg
                aria-hidden="true"
                fill="none"
                height="1em"
                viewBox="0 0 10 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M.12 1.88L6.227 8 .12 14.12 2 16l8-8-8-8L.12 1.88z"
                  fill="#32324D"
                />
              </svg>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
  <div
    class="c28"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;
